blottertrax/helper/repost_checker.py

```diff
@@ -0,0 +1,60 @@
+import sqlite3
+
+
+class RepostChecker:
+    cursor = None
+    sql = None
+
+    @classmethod
+    def __init__(self):
+        self.sql = sqlite3.connect('{}/../{}/../database/repost.db')
+        self.cursor = self.sql.cursor()
+    
+        self.cursor.execute('CREATE TABLE IF NOT EXISTS artistSubmission(artist TEXT UNIQUE NOT NULL, postTime REAL NOT NULL, repeatCount INT NOT NULL)')
+        self.cursor.execute('CREATE TABLE IF NOT EXISTS songSubmission(artistSong TEXT UNIQUE NOT NULL, postTime REAL NOT NULL)')
+        self.cursor.execute('CREATE TABLE IF NOT EXISTS idStamped(postID TEXT UNIQUE NOT NULL, postTime REAL NOT NULL)')
+        
+        self.cursor.execute('SELECT * FROM artistSubmission')
+    
+    @classmethod
+    def new_entry(self, artist_name, song_name, curr_time, postID):
+        self.cursor.execute('INSERT OR IGNORE INTO artistSubmission VALUES(?, ?, ?)', [artist_name, curr_time, 0])
+        self.cursor.execute('INSERT OR IGNORE INTO songSubmission VALUES(?, ?)', [song_name + '--||' + artist_name, curr_time])

```

Comment:
Hm I looked into this. This would be a nontrivial change, as it would require us to drop uniqueness of entries into that database, which would mean having to go about checking for existence then deleting and whatnot whenever a new post is made. It's not super hard to implement, but I'm not sure if it's really necessary, when we can just make a separator string (not like it can possibly cause an error, and even if an artist/song name somehow contains the separator string, the way it is used here means that it will still correctly identify the correct entry when a new post of the same one is posted

Reply:
the only way this could possibly produce erronous result is if for instance there is an artist called "artist --|| name" that makes song "song name", and another artist "artist" makes a song "name--||song name", which would incorrectly be identified as equal (not to mention the separator string can be made as complex as we want)

If there is _really_ a desire to do it can still be made proper, but I'm not sure if it is necessary